argoproj/argo-workflows
# fix: get configmap data when updating controller config Fixes #10659

https://github.com/argoproj/argo-workflows/issues/10659 has details of the bug.

Verification steps:

0. I checkout master
1. `kind create cluster --name reproduction`
2. Install argo workflows w/ helm `helm install argo-workflows argo/argo-workflows --set controller.workflowDefaults.metadata.labels.simple=1 --version 0.22.13` (the 0.22.13 chart is using Argo Workflows v3.4.5)
3. I deploy a workflow. I see that it does set the label value.
4. I upgrade the default `helm install argo-workflows argo/argo-workflows --set controller.workflowDefaults.metadata.labels.simple=2 --version 0.22.13`
5. I verify in the configmap that the value has changed.
6. I deploy a workflow. It sets the label to the original value (simple: 1), not the new value.
7. When I look at the logs for the argo-workflows-controller, I see it knows there is an update, but I see it prints out the old version of the configmap
8. I checkout my branch.
9. I run `make workflow-controller-image`
10. I retag the image `docker tag quay.io/argoproj/workflow-controller:latest quay.io/argoproj/workflow-controller:me`
11. I load the image into kind `kind load docker-image quay.io/argoproj/workflow-controller:me --name reproduction`
12. I deploy the image `helm upgrade argo-workflows argo/argo-workflows --set controller.workflowDefaults.metadata.labels.simple=2 --set controller.image.tag=me --version 0.22.13 --set images.pullPolicy=IfNotPresent`
13. I deploy a workflow. See that it does set the label simple to 2.
14. I update the the workflowDefaults, `helm upgrade argo-workflows argo/argo-workflows --set controller.workflowDefaults.metadata.labels.simple=3 --set controller.image.tag=me --version 0.22.13 --set images.pullPolicy=IfNotPresent`
15. I deploy a workflow. See that it does set the label simple to 3.

Fixes #10659

In terms of implementation, I could have done this in other ways. I picked this approach because it is the least code added and therefore the simplest.
